corehq/apps/reports/commtrack/data_sources.py

```diff
@@ -434,19 +434,18 @@ def get_data(self):
 
                 results = StockReport.objects.filter(
                     stocktransaction__case_id=spoint_id
+                ).filter(
+                    date__gte=self.converted_start_datetime,
+                    date__lte=self.converted_end_datetime
                 ).values_list(
                     'form_id',
                     'date'
-                ).order_by('-date').distinct()  # not truly distinct due to ordering
```

Comment:
Looks like order was only needed for the not matched case, which uses a new query